### PR TITLE
fix_: ensure individual store queries do not exceed 24h while still allowing to retrieve more than one day of messages

### LINKED ISSUE #6057: waku: ensure that fetching a community is done using 24h query batches

### PR DESCRIPTION
**Pull request by richard-ramos**
Fixes https://github.com/status-im/status-go/issues/6057

Also, due to how we use seconds in some parts of the code, there's a loss of precision in the end date of the store queries which translates in potentially missed messages. To quickly fix this I just add a second and subtract a nanosecond to ensure that the end date will include the whole second. 

This however should be properly fixed by using time.Time across all codebase instead as mentioned in https://github.com/status-im/status-go/pull/5857#discussion_r1809502325

cc: @fryorcraken

### CODE REVIEW

File: protocol/messenger_mailserver.go [hunk 1]
```diff
@@ -880,6 +855,8 @@ loop:
 					contentTopics: w.contentTopics,
 					cursor:        cursor,
 					limit:         nextPageLimit,
+					from:          w.from,
+					to:            nextWorkTo,
```
Comment (kaichaosun) [inline protocol/messenger_mailserver.go]: What happens if above line `cursor == nil`?
It seems we need to first processNextPage with the cursor then process the `nextWorkTo`? Now it combines the cursor (processNextPage) and `nextWorkTo`.

Comment (richard-ramos) [reply protocol/messenger_mailserver.go]: ah good point!

**Review by jrainville — APPROVED** [review]
LGTM.

BTW,  we found today with @Ivansete-status that some of the store requests fetched a whole month. I assume it was an account recovery, but not sure.

So I guess this PR would fix it.

One additional improvement in that case would be to make the account recovery smarter. If you fetched all the needed info in the last 24 hours, you don't need the info from a week past, since you only need the most recent.

Maybe something to fix in another issue. Though I'm not sure if it should be fixed by the Status or Waku team :thinking:

**Comment by richard-ramos** [comment]
> @jrainville: One additional improvement in that case would be to make the account recovery smarter. If you fetched all the needed info in the last 24 hours, you don't need the info from a week past, since you only need the most recent.

To fix that, we can do something similar to what @igor-sirotin implemented here: https://github.com/status-im/status-go/blob/72f7ed7ced4d7a0838874fe04c00eec0290eae86/protocol/messenger_store_node_request_manager.go#L558 In which a [function](https://github.com/status-im/status-go/blob/72f7ed7ced4d7a0838874fe04c00eec0290eae86/protocol/messenger_store_node_request_manager.go#L377-L500) is passed to `processMailserverBatchWithOptions` to indicate whether to continue fetching messages from storenode or not.

**Review by kaichaosun — APPROVED** [review]

**Review by plopezlpz — APPROVED** [review]

**Comment by kaichaosun** [comment]
This PR also fix https://github.com/status-im/status-go/issues/6046

**Pull request merged**